processing/src/main/java/io/druid/segment/StringDimensionHandler.java

```diff
@@ -120,24 +120,20 @@ public void validateSortedEncodedArrays(
               dimValName
           );
         }
+      } else {
```

Comment:
Also I think those `if (lhsLen == 0 && rhsLen == 1) {` and `} else if (rhsLen == 0 && lhsLen == 1) {` could be removed because it would be handled by the loop below anyway.